kubernetes_platform/Makefile

```diff
@@ -35,11 +35,16 @@ clean-go:
 	rm -rf kubernetes_platform/go
 
 
-# Generate Python package.
+# Build and Locally install Python package.
 .PHONY: python
 python: proto/kubernetes_executor_config.proto
 	python3 python/generate_proto.py && cd python && python3 setup.py bdist_wheel
 
+# Build and Locally install Python package using editable mode for development.
+.PHONY: python-dev
+python-dev: proto/kubernetes_executor_config.proto
+	python3 python/generate_proto.py && cd python && pip install -e .[dev]
```

Comment:
it will install extras: https://github.com/kubeflow/pipelines/blob/master/kubernetes_platform/python/setup.py#L83

pip docs, see step 7: https://pip.pypa.io/en/stable/cli/pip_install/#examples